src/tchannel_bridge.js

```diff
@@ -0,0 +1,103 @@
+        return new tchannelSpan({
+            id: Utils.tchannelBufferToIntId(span.context().spanId),
+            traceid: Utils.tchannelBufferToIntId(span.context().traceId),
+            name: span.name,
+            parentid: Utils.tchannelBufferToIntId(span.context().parentId),
+            flags: span.context().flags || 0
+        });
+    }
+
+    static toTChannelv2Span(span: Span): TChannelSpan {
+        return new tchannelSpan({
+            id: span.context().spanId,
+            traceid: span.context().traceId,
+            name: span.name,
+            parentid: span.context().parentId,
+            flags: span.context().flags || 0
+        });
+    }
+
+    static getSpanFromTChannelRequest(tracer: Tracer, request: any, headers: any = {}, options: any = {}): Span {
+        let traceContext;
+        if (crossdock_constants.TCHANNEL_HEADER_TRACER_STATE_KEY in headers) {
+            traceContext = SpanContext.fromString(headers[`${crossdock_constants.TCHANNEL_HEADER_TRACER_STATE_KEY}`]);
+        } else if (request.span) {
+            // $FlowIgnore - I just want an empty span context.
+            traceContext = new SpanContext();
+            if (Buffer.isBuffer(request.span.id)) {
+                traceContext.spanId = request.span.id;
+                traceContext.traceId = request.span.traceid;
+                traceContext.parentId = request.span.parentid;
+                traceContext.flags = request.span.flags;
+            } else {
+                traceContext.spanId = new Int64(request.span.id[0], request.span.id[1]).toBuffer();
+                traceContext.traceId = new Int64(request.span.traceid[0], request.span.traceid[1]).toBuffer();
+                traceContext.parentId = new Int64(request.span.parentid[0], request.span.parentid[1]).toBuffer();
+                traceContext.flags = request.span.flags;
+            }
+        }
+
+        options.childOf = traceContext;
+        let span = tracer.startSpan(String(request.arg1), options);
+
+        for (let key in headers) {
+            let keyWithoutTChannelPrefix = key;
+            if (Utils.startsWith(key, crossdock_constants.TCHANNEL_TRACING_PREFIX)) {
+                keyWithoutTChannelPrefix = key.substring(crossdock_constants.TCHANNEL_TRACING_PREFIX.length);
+            }
+
+            if (Utils.startsWith(keyWithoutTChannelPrefix, constants.TRACER_BAGGAGE_HEADER_PREFIX)) {
+                let baggageKey = keyWithoutTChannelPrefix.substring(constants.TRACER_BAGGAGE_HEADER_PREFIX.length);
+                let baggageValue = headers[key];
+                span.setBaggageItem(baggageKey, baggageValue);
+            }
+        }
+
+        return span;
+    }
+
+    static saveBaggageToCarrier(span: Span, carrier: any = {}): any {
+        let baggage = span.context().baggage;
+        for (let key in baggage) {
+            if (baggage.hasOwnProperty(key)) {
+                let baggageKey = `${constants.TRACER_BAGGAGE_HEADER_PREFIX}${key}`;
```

Comment:
why not `constants.TRACER_BAGGAGE_HEADER_PREFIX + key`? String extrapolation is more expensive than simple concatenation.